### PR TITLE
Docs: Update contributing guide

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

While going through the contributing guide for the first time today, I realized it needed some updates for v4. The PR also improves some wordings.

### Why is it needed?

The ensure a smooth setup experience.

### CODE REVIEW

**Review by pwizla — DISMISSED** [review]
Hey @gu-stav , thanks! I've reviewed your PR and suggested a few minor changes. Other than that, I think it looks much cleaner!

**Comment by gu-stav** [comment]
@pwizla Thanks for you feedback. I've included all of it and made even more improvements. If you find the time, I'd appreciate another round of review 🚀

**Comment by pwizla** [comment]
Thanks @gu-stav, I'll add this to my to-do list for next week.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,129 +1,116 @@
...
 ---
 
 ## Before Submitting a Pull Request
 
-The core team will review your pull request and will either merge it, request changes to it, or close it.
+The Strapi core team will review your pull request and either merge it, request changes to it, or close it.
+
+## Contribution Prerequisites
+
+- You have [Node.js](https://nodejs.org/en/) at version >= v12 and <= v16 and [Yarn](https://yarnpkg.com/en/) at v1.2.0+ installed.
```
Comment (alexandrebodin) [inline CONTRIBUTING.md:49]: To run the unit tests you need node 14 now we can increase the require version of contributing

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
A few comments but that looks nice!

File: CONTRIBUTING.md (hunk 1 above)
Comment (alexandrebodin) [reply CONTRIBUTING.md:49]: Wasn't updated :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,129 +1,116 @@
...
 ## Feature Requests
 
-Feature Requests by the community are highly encouraged. Please feel free to submit a [feature request](https://portal.productboard.com/strapi) or to upvote 👍 [an existing feature request](https://portal.productboard.com/strapi) in the ProductBoard.
+Feature Requests by the community are highly encouraged. Feel free to submit a new one or upvote an existing feature request on [ProductBoard](https://portal.productboard.com/strapi).
 
-## RFCs
+## Request For Comments (RFC)
 
-Some important changes in Strapi require some thoughts to be put into the design phase before starting working on a PR.
+Larger chunks of changes in Strapi, that might effect many users, require a thorough design phase before starting working on a PR.
```
[the remark this answers is not in the document]
Comment (gu-stav) [reply CONTRIBUTING.md]: Good suggestion. I have added a note and ran the whole document through Grammarly in 375f41d716dd086a9226c8b78ef6f06d4c3d874f

[the remark this answers is not in the document]
Comment (alexandrebodin) [reply CONTRIBUTING.md]: Good idea 👍

**Review by alexandrebodin — APPROVED** [review]
LGTM

**Pull request merged**